src/components/activities/dashboardActivity/DashboardActivity.tsx

```diff
@@ -1,24 +1,37 @@
-import React, { FunctionComponent, useState } from "react";
+import React, { FunctionComponent, useEffect, useState } from "react";
 import { connect } from "react-redux";
 import { Redirect } from "react-router";
 import PlusIcon from "../../../assets/PlusIcon";
 import SearchIcon from "../../../assets/SearchIcon";
+import { AUTH_KEY } from "../../../consts";
+import { SessionStorage } from "../../../libraries/storage/storage";
+import { setAuthenticationSuccess } from "../../../state/main/actions";
 import { IState } from "../../../types";
 import AppHeader from "../../accessories/appHeader/AppHeader";
 import Footer from "../../accessories/footer/Footer";
 import LargeButton from "../../accessories/largeButton/LargeButton";
 import "./styles.scss";
-import { IStateProps, TProps, TActivityTransitionState } from "./types";
+import { IStateProps, TProps, TActivityTransitionState, IDispatchProps } from "./types";
 
 const DashboardActivity: FunctionComponent<TProps> = ({
   userCredentials,
   newPatientRoute,
+  setAuthenticationSuccess,
   searchPatientRoute,
 }) => {
   const breadcrumbMap = {
     Dashboard: "/",
   };
 
+  useEffect(() => {
+    if (!userCredentials) {
+      const userCredentialsLocal = SessionStorage.read(AUTH_KEY);
+      if (userCredentialsLocal) {
+        setAuthenticationSuccess(userCredentialsLocal);
+      }
+    }
+  }, [userCredentials, setAuthenticationSuccess]);
+
```

Comment:
It seems to me that the `DashboardActivity` is not the right place to do this retrieval. I mean, we want to get the `AUTH_KEY` from the _sessionStorage_ if we do a refresh at any activity, not only this one. Perhaps this logic you wrote should go in an upper layer of the components tree.

There's a _higher-order component_ called `PrivateRoute`, which wraps all the activity components that should be behind the authentication wall. It checks if the app is authenticated and renders the children if so. Maybe that's the place where we should retrieve the `AUTH_KEY` and put it in the store.

Reply:
"It seems to me that the DashboardActivity is not the right place to do this retrieval. I mean, we want to get the AUTH_KEY from the sessionStorage if we do a refresh at any activity, not only this one."

You're right, I didn't see the big picture here. I'll move up this script as suggested.